src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2813,6 +2814,30 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - Moves the tab to another index in the tabs row (if required).
+    // Arguments:
+    // - currentTabIndex: the current index of the tab to move
+    // - suggestedNewTabIndex: the new index of the tab, might get clamped to fit int the tabs row boundaries
+    // Return Value:
+    // - <none>
+    void TerminalPage::_TryMoveTab(const uint32_t currentTabIndex, const int32_t suggestedNewTabIndex)
+    {
+        uint32_t newTabIndex = std::clamp<int32_t>(suggestedNewTabIndex, 0, _tabs.Size() - 1);
+        if (currentTabIndex != newTabIndex)
+        {
+            auto tab = _tabs.GetAt(currentTabIndex);
+            auto tabViewItem = tab.TabViewItem();
+            _tabs.RemoveAt(currentTabIndex);
+            _tabs.InsertAt(newTabIndex, tab);
+            _UpdateTabIndices();
```

Comment:
Does this just magically update the MRU order as well?

Reply:
@zadjii-msft - MRU contains commands rather than tabs. So it doesn't matter what is the tab position we will switch to it.

Reply:
Actually I might be wrong.. cause for some reason SwitchToTabCommand stores index rather than the tab itself.. Checking..

Reply:
Ok, it works because `_UpdateIndices`, calls `UpdateTabViewIndex` which updates the index.

Reply:
Okay, that's what I thought it might do, but I wasn't sure. Thanks for confirming!